### ISSUE TITLE
Project missing in environment

### ISSUE THREAD

**Issue by FlorianWilhelm** [juftin/hatch-pip-compile#6]
Thanks for providing [hatch-pip-compile](https://github.com/juftin/hatch-pip-compile), it looks really useful but I am having a strange problem. Somehow the project itself is not installed in the default environment if it has type `"pip-compile"` as opposed to `"virtual"`. So in my `pyproject.toml` I have:

```toml
[tool.hatch.env]
requires = ["hatch-pip-compile"]  

[tool.hatch.envs.default]
type = "pip-compile"
[tool.hatch.envs.default.scripts]
upgrade-all = "pip-compile --upgrade {args}"
upgrade-pkg = "pip-compile --upgrade-package {args}"
```

and running `hatch run python -c "import name_of_my_package"` it doesn't find the package. Also `hatch run pip list` doesn't show it. Maybe I am missing something obvious here 🤔 .

**Comment by juftin** [comment]
Hey @FlorianWilhelm thank you for reporting this, I actually noticed this the other day. I believe this is caused by using `pip-sync` underneath all of this to manage the environment. I will get a fix out to this soon (today hopefully).

**Comment by juftin** [comment]
Alright, I've gotten this resolved in [v0.2.2](https://github.com/juftin/hatch-pip-compile/releases/tag/v0.2.2). `hatch-pip-compile` is still in early days so thank you for filing an issue and making this better for everyone. Please let me know if you encounter any other issues 🤝

**Comment by FlorianWilhelm** [comment]
Thanks, that was really fast! It works.
